v2/index/index.go

```diff
@@ -47,6 +45,10 @@ type (
 		Unmarshal(r io.Reader) error
 
 		// Load inserts a number of records into the index.
+		// Note that Index will load all given records. Any filtering of the records such as
+		// exclusion of CIDs with multihash.IDENTITY code must occur prior to calling this function.
+		// Further, the actual information extracted and indexed from the given records entirely
+		// depends on the concrete index implementation.
```

Comment:
I think that last sentence deserves to be on a separate paragrah and with an example - "for example, some index implementations may only store partial multihashes".